### PR TITLE
feat(client): Allow iframe node in send function

### PR DESCRIPTION
**Pull request by trainerbill**
I first tried to use the library like this and it errored.  The docs really don't say much about iframe.contentWindow so I had to do some searching.  I think this makes sense and may save confusion.

Allows:
```
var frame = document.getElementById('myFrame');
postRobot.send(frame, 'test', data);
```

Also moved the options.window check to after all the ways to set options.window are complete.  Make sure at the very end options.window.document exists.

### CODE REVIEW

File: src/interface/client.js [hunk 1]
```diff
@@ -34,9 +30,13 @@ export function request(options) {
...
 
-            if (!options.window) {
-                throw new Error('Expected options.window');
-            }
+            
+        } else if (options.window.tagName && options.window.tagName.toLowerCase() === 'iframe') {
+            options.window = options.window.contentWindow;
+        }
+
+        if (!options.window || !options.window.document) {
```
Comment (bluepnume) [inline src/interface/client.js]: We can't check for `options.window.document` cross-domain

File: src/interface/client.js [hunk 2]
```diff
@@ -34,9 +30,13 @@ export function request(options) {
 
             options.window = el.contentWindow;
 
-            if (!options.window) {
-                throw new Error('Expected options.window');
-            }
+            
+        } else if (options.window.tagName && options.window.tagName.toLowerCase() === 'iframe') {
```
Comment (bluepnume) [inline src/interface/client.js]: If `options.window` is a cross-domain window, checking `options.window.tagName` is going to throw a cross-domain error.

**Comment by bluepnume** [comment]
Few potential cross-domain errors might be caused by this. My suggestion is, the logic should be:

```
if (mockmode) { /* use current window */ }
else if (typeof this.window === 'string') { /* Look for iframe with querySelector / getElementById */ }
else if (options.window instanceof HTMLElement) { /* Use contentWindow if iframe */ }
else if (typeof options.window === 'object') { /* Assume options.window is a window */ }
else { /* error */ }
```

File: src/interface/client.js [hunk 3]
```diff
@@ -34,9 +30,13 @@ export function request(options) {
...
+            options.window = options.window.contentWindow;
+        }
+
+        if (!options.window || !options.window.document) {
+            throw new Error('Expected options.window to be a window object.');
```
Comment (bluepnume) [inline src/interface/client.js]: We should say something like `'Expected options.window to be a window object, iframe, or iframe element id.'`

File: test/test.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ function createPopup(name) {
 
 let bridge;
 
-let childWindow, childFrame, otherChildFrame;
+let childWindow, childFrame, otherChildFrame, frameElement;
```
Comment (bluepnume) [inline test/test.js]: Test looks perfect 👍

**Comment by trainerbill** [comment]
@bluepnume PR updated with your reviews.

File: src/interface/client.js [hunk 5]
```diff
@@ -34,9 +30,22 @@ export function request(options) {
...
-            if (!options.window) {
-                throw new Error('Expected options.window');
+            
+        } else if (options.window instanceof HTMLElement) {
+            
+            if (options.window.tagName.toLowerCase() !== 'iframe') {
+                throw new Error(`Expected options.window ${options.window} to be an iframe`);
+            }
+
+            if (!options.window.contentWindow) {
```
Comment (bluepnume) [inline src/interface/client.js]: Let's also add this check when a string is passed for `options.window` to cover all of our bases?

**Comment by bluepnume** [comment]
One more comment, otherwise looks perfect

**Comment by trainerbill** [comment]
Done

**Comment by bluepnume** [comment]
Cool! Merging.

**Pull request merged**